### PR TITLE
Updating type hints

### PR DESCRIPTION
**Pull request by apchytr**
Reopening to avoid unwanted commits 

**Context:** The use of `Optional` and `Union` should be replaced with `|` according to [PEP standards](https://peps.python.org/pep-0604/). 

**Description of the Change:** `Optional` -> ` | None`, `Union` -> ` | `, `Tuple` -> `tuple`, etc

### CODE REVIEW

**Review by ziofil — APPROVED** [review]
thanks!

File: mrmustard/physics/ansatze.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def __init__(self) -> None:
         self._kwargs = {}
 
     @abstractmethod
-    def from_function(cls, fn: Callable, **kwargs: Any) -> Ansatz:
+    def from_function(cls, fn: Callable, **kwargs: any) -> Ansatz:
```
Comment (ziofil) [inline mrmustard/physics/ansatze.py:69]: shouldn't `any` be `Any`?

Comment (apchytr) [reply mrmustard/physics/ansatze.py:69]: Actually yes! You're right. It's not until now that I found out `any` is referring to the function and not the type

**Review by ziofil — APPROVED** [review]
great 👍

**Pull request merged**